File: q5.cpp
// Move all the negative elements to one side of the array i.e. to the left side
// Neat-o
// Learnt queue STL
// Code runs in O(n)
// { Driver Code Starts
#include <bits/stdc++.h>
using namespace std;


void segregateElements(int [], int );

int main() {
	// your code goes here
	int t;
	cin>>t;
	while(t--)
	{
		int n;
		cin>>n;
		int a[n];
		for(int i=0;i<n;i++)
		cin>>a[i];
		
		segregateElements(a,n);
		
        for(int i=0;i<n;i++)
        cout<<a[i]<<" ";
		cout<<endl;
	}
}
#include <queue>
void segregateElements(int arr[],int s)
{
    // Your code goes here
    int p=0,temp,j,n=-1,count=0;
    queue <int> q;
    for(p=0;p<s;p++)
    {
        if(arr[p]<0)
        {
            q.push(arr[p]);           
        }
        else
        {
            n++;
            arr[n] = arr[p];
        }        
    }
    while(!q.empty())
    {
        arr[++n] = q.front();
        q.pop();
    }
}

// 0.4

File: q5.py
# Move all the negative elements to one side of the array i.e. to the left side
#User function Template for python3

def segregateElements( arr, s):
    # Your code goes here
    n = -1
    q = []
    for i in range(s):
        if(arr[i]<0):
            q.append(arr[i])
        else:
            n = n+1
            arr[n] = arr[i]
    #print(*q)
    while(q):
        n = n+1
        arr[n] = q[0]
        q.remove(q[0])


#{ 
#  Driver Code Starts
#Initial Template for Python 3


def main():

    T = int(input())

    while(T > 0):
        n = int(input())
        a = [int(x) for x in input().strip().split()]
        segregateElements(a, n)
        print(*a)

        T -= 1


if __name__ == "__main__":
    main()


# 1.86